### DISCUSSION TITLE
Guaranteed lifetime of {input.begin(), input.end()} after an action ::apply() method

### DISCUSSION THREAD

**Discussion by 0x8000-0000** [taocpp/PEGTL#280] [Q&A] [answered]
Regarding

https://github.com/taocpp/PEGTL/blob/main/doc/Actions-and-States.md?plain=1#L200

       More importantly the action_input does not own the data it points to, it belongs to
       the original input used in the parsing run. Therefore the validity of the pointed-to
       data might not extend (much) beyond the call to apply()!

I am creating (by hand for now) a parse tree, and I would like to record some "strings" cheaply in the nodes, using the result of input.string_view(). I understand they might dangle, but if my original input comes from ``pegtl::memory_input`` constructed from a string_view() over a buffer I own, can I safely assume that as long as that buffer is alive, all my references will point into it and will not dangle? I presume the ``pegtl::memory_input`` object can be destroyed. (It would make sense for it to be so, but just checking.)

**Comment by d-frey** [comment, marked as answer]
Yes, that should work.
